### PR TITLE
Refactor templates pipelines

### PR DESCRIPTION
**Pull request by theopnv**
### Purpose of this PR

https://jira.unity3d.com/browse/SDET-220

---
### Testing status

- HDRP: https://yamato.cds.internal.unity3d.com/jobs/902-Graphics/tree/yamato%252Frefactor-templates-pipelines/.yamato%252F_templates.yml%2523publish_hdrp_template_trunk_dry_run/6637047/job
- URP: https://yamato.cds.internal.unity3d.com/jobs/902-Graphics/tree/yamato%252Frefactor-templates-pipelines/.yamato%252F_templates.yml%2523publish_universal_template_trunk_dry_run/6637040/job

^ These 2 jobs are failing because version 12.0.0 was already pushed to the candidates, but I'm confident that the publish job itself is now working.

### CODE REVIEW

**Comment by theo-pnv** [comment]
Note: Let's merge the gfx-sdet-tools companion PR before this one so we can update the gfx-sdet-tools pointer.
We can review both at the same time.

**Review by sebastienlagarde — APPROVED** [review]

File: .yamato/_templates.yml [hunk 1]
```diff
@@ -8,29 +8,33 @@
...
+pre_pack_hdrp_template_Win_trunk:
+    name: Pre-Pack HDRP Template Win trunk
     agent:
         type: Unity::VM
         image: package-ci/win10:stable
         flavor: b1.large
     commands:
-      -  npm install upm-ci-utils@stable -g --registry https://artifactory.prd.cds.internal.unity3d.com/artifactory/api/npm/upm-npm
-      -  upm-ci template pack --project-path com.unity.template-hd
+      -  pip install unity-downloader-cli --index-url https://artifactory.prd.it.unity3d.com/artifactory/api/pypi/pypi/simple --upgrade
```
Comment (sophiaaar) [inline .yamato/_templates.yml:18]: arent we moving away from using pip for unity-downloader-cli?

File: .yamato/_templates.yml [hunk 2]
```diff
@@ -43,383 +47,459 @@ test_hdrp_template_Win_trunk:
...
-      -  upm-ci template test -u .\.Editor --type updated-dependencies-tests --project-path com.unity.template-hd
+      -  cp ./upm-ci~/packages/*.tgz ./.Editor/Unity.app/Contents/Resources/PackageManager/Editor
+      -  upm-ci template test -u $(pwd)/.Editor/Unity.app/Contents/MacOS/Unity --project-path com.unity.template-hd
     artifacts:
         logs:
             paths:
               -  "**/test-results/**"
-test_universal_template_Win_trunk:
-    name: Test Universal Template Win trunk
+publish_hdrp_template_trunk:
```
Comment (sophiaaar) [inline .yamato/_templates.yml:113]: is it worth adding some dry-run publish jobs?

File: .yamato/_templates.yml (hunk 1 above)
Comment (theo-pnv) [reply .yamato/_templates.yml:18]: Thanks, I'll change that.

File: .yamato/_templates.yml (hunk 2 above)
Comment (theo-pnv) [reply .yamato/_templates.yml:113]: That would be ideal, but from the docs the `publish` command does not handle the --dry-run argument:
```
upm-ci template publish [-h]
  [--project-path PROJECT_PATH]
  [--artifacts-path ARTIFACTS_PATH]
```
That said I realize it is handled in the package context. So maybe it's just an oversight and it is available in the template context as well. Let me try

File: .yamato/_templates.yml [hunk 3]
```diff
@@ -43,383 +49,521 @@ test_hdrp_template_Win_trunk:
...
-      -  .yamato/_packages.yml#pack_hdrp
+      -  .yamato/_templates.yml#pack_hdrp_template_Win_trunk
       -  .yamato/_editor_priming.yml#editor:priming:trunk:Win
     commands:
       -  npm install upm-ci-utils@stable -g --registry https://artifactory.prd.cds.internal.unity3d.com/artifactory/api/npm/upm-npm
-      -  pip install unity-downloader-cli --index-url https://artifactory.prd.it.unity3d.com/artifactory/api/pypi/pypi/simple --upgrade
+      -  choco install unity-downloader-cli -y -s https://artifactory.prd.it.unity3d.com/artifactory/api/nuget/unity-choco-local
       -  unity-downloader-cli --source-file unity_revision.txt -c editor --wait --published-only
-      -  for /r packages_temp %%x in (*.tgz) do copy %%x upm-ci~\packages
       -  copy upm-ci~\packages\*.tgz .Editor\Data\Resources\PackageManager\Editor
```
Comment (vicky-voin) [inline .yamato/_templates.yml:58]: Just wanted to double check if this line is still necessary since the one above it has been removed?

Comment (theo-pnv) [reply .yamato/_templates.yml]: it isn't! Thanks, good catch

**Review by vicky-voin — APPROVED** [review]

File: .yamato/_templates.yml (hunk 2 above)
Comment (theo-pnv) [reply .yamato/_templates.yml:113]: It works, I guess that was just missing in the docs :) Thanks for catching that

**Review by sophiaaar — APPROVED** [review]

**Pull request merged**